pkg/logger/logger.go

```diff
@@ -0,0 +1,119 @@
+package logger
+
+import (
+	"encoding/json"
+	"fmt"
+	"os"
+)
+
+type loggerState byte
+
+const (
+	new loggerState = iota
```

Comment:
This seems to be shadowing Go's builtin `new()` function, a different name should be used. Note that I left another comment that may make this point moot.